hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/compactions/Compactor.java

```diff
@@ -533,4 +550,46 @@ protected InternalScanner createScanner(HStore store, ScanInfo scanInfo,
     return new StoreScanner(store, scanInfo, scanners, smallestReadPoint, earliestPutTs,
         dropDeletesFromRow, dropDeletesToRow);
   }
+
+  /**
+   * Default implementation for committing store files created after a compaction. Assumes new files
+   * had been created on a temp directory, so it renames those files into the actual store dir,
+   * then create a reader and cache it into the store.
+   * @param cr the compaction request.
+   * @param newFiles the new files created by this compaction under a temp dir.
+   * @param user the running user.
+   * @return A list of the resulting store files already placed in the store dir and loaded into the
+   * store cache.
+   * @throws IOException if the commit fails.
+   */
+  public List<HStoreFile> commitCompaction(CompactionRequestImpl cr, List<Path> newFiles, User user)
+      throws IOException {
+    List<HStoreFile> sfs = new ArrayList<>(newFiles.size());
+    for (Path newFile : newFiles) {
+      assert newFile != null;
+      this.store.validateStoreFile(newFile);
+      // Move the file into the right spot
+      HStoreFile sf = createFileInStoreDir(newFile);
+      if (this.store.getCoprocessorHost() != null) {
+        this.store.getCoprocessorHost().postCompact(this.store, sf, cr.getTracker(), cr, user);
+      }
+      assert sf != null;
+      sfs.add(sf);
+    }
+    return sfs;
+  }
+
+  /**
+   * Assumes new file was created initially on a temp folder.
```

Comment:
Don't we want the Compactor asking the StoreEngine to do the writing for us? Rather than doing it here?

Reply:
We still need to vary the logic of "where" the store file is originally created. Default behaviour is to create it under temp (that's done by the call from lines #591/#592), whilst `DirectStoreCompactor` assumes the passed path is already in the store dir itself, so it just needs to create the reader over the passed path.

Reply:
Originally, the logic of creating files in temp dir was already implemented in the abstract `Compactor`, not on `DefaultCompactor`. That's why I left it here. I can leave this method abstract here and move the implementation to `DefaultCompactor`, though.

Reply:
Actually, when I tried to move it out of `Compactor` by making it abstract, it immediately  broke `StripeCompactor` and `DateTieredCompactor`, which are relying in this logic, but not extending `DefaultCompactor`.

Reply:
> it immediately broke StripeCompactor and DateTieredCompactor, which are relying in this logic, but not extending DefaultCompactor.

Devil's advocate: do you think that's correct for StripeC and DTC to do?

Reply:
I honestly had not looked further into these two. Am only worried to not break current logic used by those. Was planning to take care of those later on another PR, if we are to make it compatible with direct store/renameless writing.